controllers/activemqartemis_reconciler.go

```diff
@@ -2496,26 +2546,36 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) NewStatefulSetForCR(customResou
 func (reconciler *ActiveMQArtemisReconcilerImpl) NewPersistentVolumeClaimArrayForCR(customResource *brokerv1beta1.ActiveMQArtemis, namer common.Namers, arrayLength int) *[]corev1.PersistentVolumeClaim {
 
 	var pvc *corev1.PersistentVolumeClaim = nil
+
 	capacity := "2Gi"
 	pvcArray := make([]corev1.PersistentVolumeClaim, 0, arrayLength)
 	storageClassName := ""
 
-	namespacedName := types.NamespacedName{
-		Name:      customResource.Name,
-		Namespace: customResource.Namespace,
-	}
+	if customResource.Spec.DeploymentPlan.PersistenceEnabled {
 
-	if customResource.Spec.DeploymentPlan.Storage.Size != "" {
-		capacity = customResource.Spec.DeploymentPlan.Storage.Size
-	}
+		namespacedName := types.NamespacedName{
+			Name:      customResource.Name,
+			Namespace: customResource.Namespace,
+		}
+
+		if customResource.Spec.DeploymentPlan.Storage.Size != "" {
+			capacity = customResource.Spec.DeploymentPlan.Storage.Size
+		}
+
+		if customResource.Spec.DeploymentPlan.Storage.StorageClassName != "" {
+			storageClassName = customResource.Spec.DeploymentPlan.Storage.StorageClassName
+		}
 
-	if customResource.Spec.DeploymentPlan.Storage.StorageClassName != "" {
-		storageClassName = customResource.Spec.DeploymentPlan.Storage.StorageClassName
+		for i := 0; i < arrayLength; i++ {
+			pvc = persistentvolumeclaims.NewPersistentVolumeClaimWithCapacityAndStorageClassName(namespacedName, capacity, namer.LabelBuilder.Labels(), storageClassName, []corev1.PersistentVolumeAccessMode{"ReadWriteOnce"})
+			reconciler.applyTemplates(pvc)
+			pvcArray = append(pvcArray, *pvc)
+		}
 	}
 
-	for i := 0; i < arrayLength; i++ {
-		pvc = persistentvolumeclaims.NewPersistentVolumeClaimWithCapacityAndStorageClassName(namespacedName, capacity, namer.LabelBuilder.Labels(), storageClassName)
-		reconciler.applyTemplates(pvc)
+	for _, epvc := range customResource.Spec.DeploymentPlan.ExtraVolumeClaimTemplates {
+		nn := types.NamespacedName{Name: epvc.Name, Namespace: customResource.Namespace}
+		pvc = persistentvolumeclaims.NewPersistentVolumeClaim(nn, &epvc.Spec)
```

Comment:
The function `persistentvolumeclaims.NewPersistentVolumeClaim` ignores the annotations and labels in epvc, the new parameters could be the namespace and the epvc, i.e.
```suggestion
		pvc = persistentvolumeclaims.NewPersistentVolumeClaim(customResource.Namespace, epvc)
```

Reply:
+1